pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2466,29 +2462,45 @@ private String receiveCommandStatus() throws IOException {
   private void interpretCommandStatus(String status, ResultHandler handler) {
     long oid = 0;
     long count = 0;
-    Matcher matcher = COMMAND_COMPLETE_PATTERN.matcher(status);
-    if (matcher.matches()) {
-      // String command = matcher.group(1);
-      String group2 = matcher.group(2);
-      String group3 = matcher.group(3);
+
+    int marker1 = -1;
+    int marker2 = -1;
+    // This code processes the CommandComplete (B) message.
+    // Status is in the format of "COMMAND OID ROWS" where both 'OID' and 'ROWS' are optional
+    // and COMMAND can have spaces within it, like CREATE TABLE.
+    // Scan backwards, while searching for a maximum of two number groups
```

Comment:
Isn't `lastIndexOf(' ')` good enough?

Reply:
This has the advantage of checking for non numeric characters before parsing. So this would work correctly even with a command with space which ends in a number. Perhaps no such thing will ever exist.